### PR TITLE
Rebase to alpine 3.17 with php8.1

### PR DESCRIPTION
**Pull request by nemchik**
Reasonably significant rearranging of the init here.

* We had `root/etc/cont-init.d/60-memcache` which was actually just copying a config file (moved to `root/etc/s6-overlay/s6-rc.d/init-nextcloud-config/run`)
* We had nextcloud install and crontabs being setup in the same file (install is now in with `root/etc/s6-overlay/s6-rc.d/init-nextcloud-config/run` and crontabs is its own init)
* We had `root/defaults/nextcloud-perms.sh` which wasn't being used at all (removed)
* We had `NEXTCLOUD_PATH` set in the Dockerfile and used later in `root/defaults/nextcloud-perms.sh` (removed) and the install init (opted to ditch the variable, it's never changed and not something documented as used by nextcloud itself).

### CODE REVIEW

**Comment by thespad** [comment]
Should probably do some real-world tests of this on existing installs.

**Comment by nemchik** [comment]
I tested upgrading my personal instance and so far nothing seems to have gone wrong. I don't use nextcloud much for anything except photo backups from my phone though.

File: root/etc/s6-overlay/s6-rc.d/init-nextcloud-config/run [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+if [[ ! -f /config/www/nextcloud/config/config.php ]]; then
+    cp /defaults/config.php /config/www/nextcloud/config/config.php
+fi
+
+# permissions
+lsiown abc:abc \
+    /config \
+    /config/www/nextcloud/config/config.php \
+    /data
+lsiown -R abc:abc \
```
Comment (thespad) [inline root/etc/s6-overlay/s6-rc.d/init-nextcloud-config/run:28]: We already do this here https://github.com/linuxserver/docker-baseimage-alpine-nginx/blob/master/root/etc/s6-overlay/s6-rc.d/init-permissions/run

Comment (nemchik) [reply root/etc/s6-overlay/s6-rc.d/init-nextcloud-config/run]: Resolved in https://github.com/linuxserver/docker-nextcloud/pull/282/commits/fc101423b06f15d5da068f85a7ce6cff24c57b13

**Comment by nemchik** [comment]
Merge conflicts resolved. Nothing else really changed.

File: root/etc/s6-overlay/s6-rc.d/init-nextcloud-config/run [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+# permissions
+lsiown abc:abc \
+    /config/www/nextcloud/config/config.php \
+    /data
+
+if (occ app:list --no-interaction | grep -q richdocumentscode) 2>/dev/null; then
+    echo "Removing CODE Server"
+    occ app:remove --no-interaction richdocumentscode 2>/dev/null
+fi
```
Comment (thespad) [inline root/etc/s6-overlay/s6-rc.d/init-nextcloud-config/run:31]: What is this bit for?

**Review by thespad — APPROVED** [review]

File: root/etc/s6-overlay/s6-rc.d/init-nextcloud-config/run (hunk 2 above)
Comment (aptalca) [reply root/etc/s6-overlay/s6-rc.d/init-nextcloud-config/run:31]: The built in collabora addon that gets installed by the wizard by default. It doesn't work and causes other issues

**Pull request merged**